pyiceberg/table/__init__.py

```diff
@@ -1064,6 +1064,119 @@ def name_mapping(self) -> Optional[NameMapping]:
         """Return the table's field-id NameMapping."""
         return self.metadata.name_mapping()
 
+    def merge_rows(self, df: pa.Table, join_cols: list
```

Comment:
What do you of calling this `merge` instead? I think this aligns better with `append` and `overwrite`.
```suggestion
    def merge(self, df: pa.Table, join_cols: list
```

Reply:
@Fokko i chose "merge_rows" because when i looked through the table init.py file, i saw a lot of areas that had "merge" in it already e.g. "manifest_merge_enabled", "merge_append()", etc.

I figured it would be better to be more verbose in this case to say "merge_rows" vs. just "merge" as that might confuse others later when the look through the code.

Do you agree with this or do you still think we should change the function name to "merge" ?

Reply:
Let's get more opinions :) what do you think @kevinjqliu @bitsondatadev ? :)

Reply:
We could call it upsert, then handle full merge functionality later?

Reply:
I'm fine calling it upsert; i have no emotional ties to "merge_rows" 😂

Reply:
I think upsert is a good one as well 👍 I just don't like `rows` in the function-name

Reply:
def name has been changed from "merge_rows" to "upsert". corresponding file names have also been changed accordingly